src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -445,6 +470,45 @@ public void unregister() {
     // Subclasses may do things.
   }
 
+  @NotNull
+  public Map<String, String[]> getCourseReplInitialCommands() {
+    return replInitialCommands;
+  }
+
+  @NotNull
+  protected static Map<String, String[]> getCourseReplInitialCommands(
+      @NotNull JSONObject jsonObject,
+      @NotNull String source)
+      throws MalformedCourseConfigurationFileException {
+    Map<String, String[]> replInitialCommands = new HashMap<>();
+    JSONObject replInitialCommandsJsonObject;
+    try {
+      replInitialCommandsJsonObject = jsonObject
+          .getJSONObject("repl").getJSONObject("initialCommands");
+
+      if (replInitialCommandsJsonObject == null) {
+        return replInitialCommands;
+      }
```

Comment:
`getJSONObject` throws `JSONException` so `replInitialCommandsJsonObjet` can't possibly be `null` I think. If we want to make it optional, then I would do

```
JSONObject replJsonObject = jsonObject.optJSONObject("repl");
if (replJsonObject == null) {
  return replInitialCommands;
}
JSONObject initialCommandsJsonObject = replJsonObject.optJsonObject("initialCommands");
if (initialCommandsJsonObject == null) {
  return replInitialCommands;
}
...
```

Reply:
`null instanceOf ...` returns `false` :nerd_face: